app/cdap/components/RecipeManagement/RecipeForm/index.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { FormControl } from '@material-ui/core';
+import {
+  CancelButton,
+  CreateRecipeFormButtonWrapper,
+  EditRecipeFormButtonWrapper,
+  ErrorLabel,
+  ErrorTextField,
+  FormFieldWrapper,
+  Label,
+  NormalTextField,
+  SaveButton,
+  StyledTextAreaAutosize,
+} from 'components/RecipeManagement/RecipeForm/styles';
+import { IRecipeFormProps } from 'components/RecipeManagement/types';
+import T from 'i18n-react';
+import React, { FormEvent } from 'react';
+import { ActionType } from 'components/RecipeList/types';
+
+const PREFIX = 'features.WranglerNewUI.RecipeForm.labels';
+
+export default function RecipeForm({
+  recipeFormData,
+  isRecipeNameError,
+  recipeNameErrorMessage,
+  onRecipeNameChange,
+  onFormSubmit,
+  onCancel,
+  isSaveDisabled,
+  recipeFormAction,
+  onRecipeDescriptionChange,
+}: IRecipeFormProps) {
+  const StyledLabel = isRecipeNameError ? ErrorLabel : Label;
+  const StyledTextField = isRecipeNameError ? ErrorTextField : NormalTextField;
+  const StyledFormButtonWrapper =
+    recipeFormAction === ActionType.CREATE_RECIPE
+      ? CreateRecipeFormButtonWrapper
+      : EditRecipeFormButtonWrapper;
+
+  return (
+    <>
+      <form
+        onSubmit={(event: FormEvent<HTMLFormElement>) => onFormSubmit(event)}
+        data-testid="recipe-form-parent"
+      >
+        <FormFieldWrapper>
+          <StyledLabel data-testid="recipe-form-name-label" component="label">
+            {recipeFormAction === ActionType.CREATE_RECIPE &&
+              T.translate(`${PREFIX}.createRecipeNameLabel`)}
+            {!(recipeFormAction === ActionType.CREATE_RECIPE) &&
+              T.translate(`${PREFIX}.editRecipeNameLabel`)}
+          </StyledLabel>
+          <StyledTextField
+            autoFocus={true}
+            aria-label="recipe form recipe name textfield"
```

Comment:
This is not the correct usage of `aria-label`

Reply:
changed aria-labels with Recipe Name and Recipe Description for respective fields